File: Joseffe Barroso/Java/Polimorfismo/src/Model/ContaPoupanca.java
package Model;

public class ContaPoupanca extends Conta {
	private Double rentabilidade;

    public Double getRentabilidade() {
        return rentabilidade;
    }

    public void setRentabilidade(Double rentabilidade) {
        this.rentabilidade = rentabilidade;
    }
    
    public void depositar(Double valor) {
    	super.depositar(valor + (valor*this.rentabilidade));
    }
    
    public ContaPoupanca() {
    	
    }
   
    public ContaPoupanca(String agencia, String numero, Double salario, Double rentabilidade) {
        super(agencia, numero, salario);
       
        this.rentabilidade = rentabilidade;
    }
}

File: Joseffe Barroso/Python/Exercícios - POO/Herança/Banco/ContaPoupanca.py
from Conta import Conta

class ContaPoupanca(Conta):
    def __init__(self, nome, numConta, saldo=0, rentabilidade=0.1):
        super().__init__(nome, numConta, saldo)
        self.__rentabilidade = rentabilidade

    @property
    def rentabilidade(self):
        return self.__rentabilidade

    @rentabilidade.setter
    def limite(self, rentabilidade):
        self.__rentabilidade = rentabilidade

    def depositar(self, valor):
        valor = valor * (1 + self.__rentabilidade)
        super().depositar(valor)